### ISSUE TITLE
FormattedDate (and Time) doesn't follow locale format (Chrome)

### ISSUE THREAD

**Issue by 1someone** [yahoo/react-intl#846]
Hi,
I'm not sure this is an issue with react-intl or if it is an pure browser issue.

In my project I have a component that just renders a FormattedTime like 

`<FormattedTime value={new Date()} />`

This will render the time as text inside a span. Fine.

I have also added two locales to my application (en and se) and functionality to switch between them causing affected components to rerender themselves with current locale formatting.

```
import en from 'react-intl/locale-data/en';
import se from 'react-intl/locale-data/se';
addLocaleData([...en, ...se]);

<IntlProvider messages={this.props.translations} locale={this.props.locale}>
..... My app .....
</IntlProvider>

```
This all works fine except for time and dates in Chrome.
Eg. a time of 10:00 PM will render as

Locale en:
Chrome: 10:00 PM (Correct)
Edge: 10:00 PM (Correct)
Firefox: 10:00 PM (Correct)

Switching to locale se:
Chrome: 10:00 PM (**Wrong**)
Edge: 22:00 (Correct)
Firefox: 22:00 (Correct)

Doesn't Chrome interpret date/time formats the same way as other browsers?

Regards

**Comment by ericf** [comment]
This seems like a Chrome issue and how they decide when to set `hour12: false`. If you only have a few locales, you might want to keep a set of locales you want to use 24 hours and set `hour12={false}` https://github.com/yahoo/react-intl/wiki/API#date-formatting-apis

**Comment by 1someone** [comment]
Thanks for fast reply!

I tried to set formats on IntlProvider like so:

```
const formats = { time: { hour12: false } };
<IntlProvider messages={this.props.translations} locale={this.props.locale} formats={formats} }>
..... My app .....
</IntlProvider>
```

But it didn't have any effect. Also tried setting the defaultFormats with no luck. 
Am I doing it wrong?

Regards

**Comment by ericf** [comment]
The `formats` feature allows you to _name_ a set of options.
```js
const formats = {
  time: {
    '24hour': {hour12: false, hours: 'numeric', minute: 'numeric'}
  }
};

<IntlProvider locale={locale} messages={messages} formats={formats}>
  <FormattedTime value={someDate} format='24hour'/>
</IntlProvider>
```

Once you've defined a named format, you can then use it via the `format` prop on `<FormattedTime>`.

**Comment by 1someone** [comment]
Thanks again for excellent support!
However it turns out I have myself to blame.

Somehow I had the locales mixed up. I had added both en and se (which I thought was Swedish) and then on first render I set the locale to 'sv' on IntlProvider. this somehow caused Chrome not to change the format.

When I replaced the se with sv it all worked!

At least I learned about the `formats` feature and maybe this can help somebody from making the same mistake.

Very happy with this library so far!

Regards
